### PR TITLE
Adds mitocholide; a semisynthetic C3 organ healer.

### PR DESCRIPTION
**Pull request by Krysonism**
## About The Pull Request
This PR adds a new reagent; mitocholide. This reagent slowly heals all your organs!

The name and effect might be familiar to those who have played on non-/tg/ servers.

It can also be applied directly to an organ to heal it.

## Why It's Good For The Game
This PR has 3 main goals:

Provide a method of healing organs that lack a specific reagent such as the heart and stomach.

Increase the cooperation between botany and chemistry, chemistry has not really needed anything from botany in a long time since rezadone is not really very useful anymore and omnizine is abundant and available roundstart.

Make non-cloning revival methods more appealing by providing a method to reverse multi organ decay.

The recipe needs both polypyrylium  oligomers and omnizine as well as formaldehyde, making it the hardest medicine to make in the game, but lacks obvious direct powergaming potential.

## Changelog
:cl:
add: Mitocholide, a new multipurpose organ healer.
/:cl:

### CODE REVIEW

**Comment by wesoda25** [comment]
I highly doubt this PR will be merged. There should be specific chems to address specific traumas.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 1]
```diff
@@ -1226,7 +1226,43 @@
...
+	M.adjustStaminaLoss(1) //heart
+	if(prob(50))
+		M.Dizzy(2) //brain
+	for(var/datum/reagent/medicine/R in M.reagents.reagent_list)
+		M.reagents.remove_reagent(R.type,1) //liver
+	if(prob(5))
+		to_chat(M, "<span class='danger'>Your stomach rumbles!</span>")
+		M.adjust_disgust(10) //stomach
+	if(prob(8))
+		to_chat(M, "<span class='danger'>Your ears start ringing!</span>") //ears
```
Comment (Tlaltecuhtli) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: ring but no deaf?

**Comment by Ratvar** [comment]
This **is** a specific trauma: post-mortem organ degradation

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 1 above)
Comment (Krysonism) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: I wanted one effect for each major organ but i felt that adding deafness was too much. 

I am certainly willing to add a little bit deafness as well if you feel that it is appropriate.

**Comment by Krysonism** [comment]
> 
> 
> I highly doubt this PR will be merged. There should be specific chems to address specific traumas.

I think it would be interesting to have a "omnizine for the organs",  convenience at a premium you know. 

I think that organ damage that falls short of causing organ failure will be changed to be more debilitating in the future, currently you can have high lung and heart damage without being inconvenienced. 

If organ damage becomes more common and impactful we will need good treatment options.

I agree that this chem should not be as easy to obtain as on some other servers.

**Comment by Detective-Google** [comment]
The name puts me off of this, Everytime I read it I can only think "midichlorians"

**Comment by wesoda25** [comment]
> > I highly doubt this PR will be merged. There should be specific chems to address specific traumas.
> 
> I think it would be interesting to have an "omnizine for the organs", convenience at a premium you know.
> 
> I think that organ damage that falls short of causing organ failure will be changed to be more debilitating in the future, currently you can have high lung and heart damage without being inconvenienced.
> 
> If organ damage becomes more common and impactful we will need good treatment options.

I don’t think we should have a safe, cure all that allows you to ignore the new chem system and its drawbacks, crafting irrelevant. It effectively negates the progress (if you can call it that) we’ve made up to this point. Specific chems with tricky OD rates and varying results would be better.

**Comment by Krysonism** [comment]
> I don’t think we should have a safe, cure all that allows you to ignore the new chem system and its drawbacks, crafting irrelevant. It effectively negates the progress (if you can call it that) we’ve made up to this point. Specific chems with tricky OD rates and varying results would be better.

This chem doesn't really outheal any source of organ damage. OD rates don't really matter unless the doctor is retarded, thializid being the exception.

I would like to point out again that this is probably the hardest medicine to make in the game.

I would probably want to see the organ self healing removed from the vital organs at some point.

**Comment by carlarctg** [comment]
this medicine will never ever be made ingame due to the rarity of its chemicals

**Comment by Krysonism** [comment]
> 
> 
> this medicine will never ever be made ingame due to the rarity of its chemicals

I think it will be made whenever there is at least one chemist interested in helping medbay.

Not every round, but i know that i personally will make it.

**Comment by Krysonism** [comment]
I am thinking of adding a reaction_obj that restores damage if the chem is injected directly into an organ.

Thoughts?

**Comment by carlarctg** [comment]
MAYBE a chemist would be interested in making it, they'd take one look at the ingredients and say 'nah let's make meth'
even if they try to make it botany will probably just ignore them

**Comment by dr-bee** [comment]
Could you perhaps make it only require botany? for example have it require Silibinin instead of Formaldehyde. Having it require both chemistry and botany to make basically means doctors will never see it. As chemists have absolutely no incentive to make anything to help doctors anyway.

**Comment by Krysonism** [comment]
On Terry at least one chemist is making medicine pretty much every round

 This is concept of the both chemist AND CMO being totally incompetent is pretty foreign to me at this point in the game.

**Comment by Krysonism** [comment]
>As chemists have absolutely no incentive to make anything to help doctors anyway.

They have an incentive to not get demoted and replaced.

**Comment by Shadowflame909** [comment]
Not in the escalation days Kryonsism. The chemist with the clf3 grenade wants to see you try.

**Comment by Krysonism** [comment]
> 
> 
> Not in the escalation days Kryonsism. The chemist with the clf3 grenade wants to see you try.

If you can't manage to baton your own shithead chemist and drag him out PDA the HoS. 

I believe we should not be forced to design to the game to run smoothly when a critical subdepartment is being total shitters. Job ban them instead. We know chemists can do their job, they sure do when i am playing CMO.

**Comment by optimumtact** [comment]
I don't want organs to be chemically healable.

**Comment by Krysonism** [comment]
> 
> 
> I don't want organs to be chemically healable.

Too late?

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 1 above)
Comment (ATH1909) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: Please do so, yeah.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 2]
```diff
@@ -1226,7 +1226,53 @@
...
+
+/datum/reagent/medicine/mitocholide/on_mob_life(mob/living/carbon/M)
+	M.adjustOrganLoss(ORGAN_SLOT_LUNGS, -0.5)
+	M.adjustOrganLoss(ORGAN_SLOT_HEART, -0.5)
+	M.adjustOrganLoss(ORGAN_SLOT_BRAIN, -0.5)
+	M.adjustOrganLoss(ORGAN_SLOT_LIVER, -0.5)
+	M.adjustOrganLoss(ORGAN_SLOT_STOMACH, -0.5)
+	M.adjustOrganLoss(ORGAN_SLOT_EYES, -0.5)
+	M.adjustOrganLoss(ORGAN_SLOT_EARS, -0.5)
+	//If organ traumas are implemented I suggest giving this a chance to cure low level trauma.
```
Comment (ATH1909) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: you could make it cure low-level brain traumas, since we have those already

**Comment by ATH1909** [comment]
To be honest, I feel like this chem is a bit TOO hard to make. No one will ever make it unless it shows up in strange seeds, and even if it does, nobody will remember to USE it once they have it.

The least you could do is replace the formaldehyde in its recipe with a chem that can be easily made by botany, like vitamins or something.

**Comment by MCterra10** [comment]
The mitochondria is the powerhouse of the cell

**Comment by optimumtact** [comment]
undo whatever chems heal organs NOW

**Comment by optimumtact** [comment]
Just so we're clear, organs are specifically supposed to be healed via surgery

I don't like the trend that everyone is doing of making every single issue in the game solved by chemistry/reagents just because it happens to be one of the most fleshed out systems in the game.

im perfectly happy for surgery's to be added that heal organs, or consume reagents on touch to the organ to heal it, before you put it back in.

But I dont' want straight organ healing via chems because the surgery system needs fleshing out too

**Comment by ExcessiveUseOfCobblestone** [comment]
Also any new chems should use the suffixes (or generate their own if necessary). Idc what you set the root but I do care that all chems that heal a particular category of damage share the same suffix.

**Comment by Krysonism** [comment]
I would rather get mesothelioma than be forced to use the nonsensical and aesthetically bankrupt naming scheme you have cobbled together.

**Comment by PKPenguin321** [comment]
Heheheh, "cobbled"

**Comment by ExcessiveUseOfCobblestone** [comment]
like I said you can make your own root and even suffix if the damage type doesn't have one yet.

Heck, i'll even let you change the suffix of combo healers since I would rather have a set "combo" suffix instead of conglomerating them together (instabitaluri is a mouthful!). You can also add brand names if you'd like as long as the """chemical""" name is present as seen with synthflesh.

If you're still not willing then idk what to tell you. I'm not merging new meds that aren't using the appropriate suffix if one already exists.

**Comment by MCterra10** [comment]
>not merging a pr because they didn't use your ugly naming scheme
Name your chems in a way that makes sense, I personally prefer chems to have names based on real world chemistry and their chemical composition i.e. propionic acid

**Comment by MCterra10** [comment]
All these new medical changes are objectively terrible and I'm yet to meet one person who approves of them other than yourself and oranges, I'm glad I reverted this stuff

**Comment by optimumtact** [comment]
listen bud, if all you are here to do is brag about your downstream, you can get banned.

**Comment by ExcessiveUseOfCobblestone** [comment]
> > not merging a pr because they didn't use your ugly naming scheme
> 
> Name your chems in a way that makes sense, I personally prefer chems to have names based on real world chemistry and their chemical composition i.e. propionic acid

You're free to name them chemical compositions/metabolic pathways/etc. when you also translate those into the game so those are tangible to players. Hell, show me the chemical formula of "mitocholide" whenever you have some free time.

Until then, the names should be built off something a player could use from a game perspective. I've decided that should be what type of damage the medicine heals, and am only asking for you to retain the suffix.

**Comment by optimumtact** [comment]
also you can't have organ heals by chems as I said

**Comment by ExcessiveUseOfCobblestone** [comment]
yes that too

**Pull request closed without merging**